analysis/Scripts/SplitScript.py

```diff
@@ -32,10 +39,10 @@
     if particle == "Xic":
         mass_cuts = "lcplus_MM > 2375"
     for ybin in ybins:
-        for ptbin in ptbins:
-            yptcut = "lcplus_PT >= {0} && lcplus_PT < {1} && lcplus_RAPIDITY >= {2} && lcplus_RAPIDITY < {3}".format(ptbin[0], ptbin[1], ybin[0], ybin[1])
+        for pbin in pbins:
+            yptcut = "lcplus_PT >= {0} && lcplus_PT < {1} && lcplus_RAPIDITY >= {2} && lcplus_RAPIDITY < {3}".format(pbin[0], pbin[1], ybin[0], ybin[1])
             allcuts = "{0} && {1} && {2}".format(globalCuts, yptcut, mass_cuts)
-            wfile = TFile.Open(directory + particle + "_splitfile_y{0}-{1}_pt{2}-{3}.root".format(ybin[0],ybin[1],ptbin[0],ptbin[1]), "RECREATE")
+            wfile = TFile.Open(directory + particle + "_splitfile_y{0}-{1}_p{2}-{3}.root".format(ybin[0],ybin[1],pbin[0],pbin[1]), "RECREATE")
             subtree = alldata.CopyTree( allcuts )
```

Comment:
The issue here is that you get two copies of the DecayTree in your new TFile, correct? And they don't seem to have the same Nentries (i.e. a copy _before_ the cut, and one after, is written). That's a nuisance, and a wonderful example of crappy memory management. Could you try to swap the order of opening the TFile and declaring the new TTree? i.e. 

```suggestion
            subtree = alldata.CopyTree( allcuts )
            wfile = TFile.Open(directory + particle + "_splitfile_y{0}-{1}_p{2}-{3}.root".format(ybin[0],ybin[1],pbin[0],pbin[1]), "RECREATE")
            wfile.cd()
            subtree.Write()
```

If that does not seem to work, try opening another file to write to before the loop, where ROOT might dump it's trees without asking, and then opening our new file etc. in the loop, just like you do here?

Reply:
As I said, I checked the effect (it's a discussion within the b2mumu group as well), and it's all OK. Just don't call it with the number, just the name 'DecayTree'.